navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -260,6 +309,37 @@ private static ObjectNode putInstruction(PointList points, InstructionList instr
         Instruction instruction = instructions.get(instructionIndex);
         ArrayNode intersections = instructionJson.putArray("intersections");
 
+        // make pointList writeable
+        PointList pointList = instruction.getPoints().clone(false);
+
+        if (instructionIndex == 0) {
+             // Fix first (departure) instruction
+             // to only have one "out" intersection
+            intersectionDetails = fixFirstIntersectionDetail(intersectionDetails);
+        }  else if (instructionIndex < instructions.size() - 1) {
+            // modify pointlist to include the first point of the next instruction
+            // for all instructions but the arrival
+            PointList nextPoints = instructions.get(instructionIndex + 1).getPoints();
+            pointList.add(nextPoints.getLat(0), nextPoints.getLon(0), nextPoints.getEle(0));
+        } else {
+            // we are at the arrival (or via point arrival instruction)
+            // Duplicate the last point in the arrival instruction, which does has only one point
+            pointList.add(pointList.getLat(0), pointList.getLon(0), pointList.getEle(0));
+
+            // Add an arrival intersection with only one enty
+            ObjectNode intersection = intersections.addObject();
+            ArrayNode entryArray = intersection.putArray("entry");
+            entryArray.add(true);
+
+            // copy the bearing from the previous instruction
+            ArrayNode bearingsrray = intersection.putArray("bearings");
+            bearingsrray.add( 0);
+
+            // add the in tag
+            intersection.put("in", 0);
+            putLocation(pointList.getLat(0), pointList.getLon(0), intersection);
```

Comment:
Shouldn't this also be part of the method `fixFirstIntersectionDetail` and then the method should be renamed? This feels quite unexpected to me.